src/pk.c

```diff
@@ -507,12 +507,10 @@ static int der_to_enc_pem_alloc(unsigned char* der, int derSz,
     byte* cipherInfo = NULL;
     int pemSz = 0;
     int hashType = WC_HASH_TYPE_NONE;
-#if !defined(NO_SHA256)
-    hashType = WC_SHA256;
+#if !defined(NO_MD5)
+    hashType = WC_MD5;
 #elif !defined(NO_SHA)
```

Comment:
Any reason we can't have a SHA256 option here too?

Reply:
Regarding to [RFC8018(section 5.1)](https://datatracker.ietf.org/doc/html/rfc8018#section-5.1), PBKDF1 supports `MD2`, `MD5` or `SHA-1`, but not `SHA-2`. Therefore, I remove `SHA256`.